bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/StationHandler.java

```diff
@@ -0,0 +1,223 @@
+                });
+                ThingBuilder thingBuilder = editThing();
+                thingBuilder.withProperties(properties);
+                updateThing(thingBuilder.build());
+            } catch (SncfException e) {
+                this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    private void scheduleRefresh(ZonedDateTime when) {
+        long wishedDelay = ZonedDateTime.now().until(when, ChronoUnit.SECONDS);
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            long existingDelay = job.getDelay(TimeUnit.SECONDS);
+            if (existingDelay < wishedDelay) {
+                return;
+            }
+            freeRefreshJob();
+        }
+        refreshJob = scheduler.schedule(this::queryApiAndUpdateChannels, wishedDelay, TimeUnit.SECONDS);
+    }
+
+    private void queryApiAndUpdateChannels() {
+        SncfBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            try {
+                bridgeHandler.getNextPassage(stationId, GROUP_DEPARTURE).ifPresentOrElse(departure -> {
+                    getThing().getChannels().stream().map(Channel::getUID).filter(
+                            channelUID -> isLinked(channelUID) && GROUP_DEPARTURE.equals(channelUID.getGroupId()))
+                            .forEach(channelUID -> {
+                                State state = getValue(channelUID.getIdWithoutGroup(), departure, GROUP_DEPARTURE);
+                                updateState(channelUID, state);
+                            });
+                    scheduleRefresh(fromDTO(departure.stopDateTime.departureDateTime));
+                }, () -> {
+                    logger.debug("No {} available", GROUP_DEPARTURE);
+                    scheduleRefresh(ZonedDateTime.now().plusMinutes(5));
+                });
+                bridgeHandler.getNextPassage(stationId, GROUP_ARRIVAL).ifPresentOrElse(arrival -> {
+                    getThing().getChannels().stream().map(Channel::getUID)
+                            .filter(channelUID -> isLinked(channelUID) && GROUP_ARRIVAL.equals(channelUID.getGroupId()))
+                            .forEach(channelUID -> {
+                                State state = getValue(channelUID.getIdWithoutGroup(), arrival, GROUP_ARRIVAL);
+                                updateState(channelUID, state);
+                            });
+                    scheduleRefresh(fromDTO(arrival.stopDateTime.arrivalDateTime));
+                }, () -> {
+                    logger.debug("No {} available", GROUP_ARRIVAL);
+                    scheduleRefresh(ZonedDateTime.now().plusMinutes(5));
+                });
+                updateStatus(ThingStatus.ONLINE);
+            } catch (SncfException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                freeRefreshJob();
```

Comment:
That means in case of exception, you will then never check again ?
I don't know if it is or not the expected behaviour.

Reply:
Well, if there is a communication error, I think the user has to figure whats going on and reactivate the thing. I could reschedule in a few minutes...WDYT ?

Reply:
Ok, let's keep it like that.